src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -159,6 +165,20 @@ public SnowflakeSinkServiceV2(
             .getClient(this.connectorConfig);
     this.enableSchematization = enableSchematization;
     this.partitionsToChannel = partitionsToChannel;
+
+    this.tableName2SchemaEvolutionPermission = new HashMap<>();
+    if (this.topicToTableMap != null) {
+      this.topicToTableMap.forEach(
+          (topic, tableName) -> {
+            if (!this.tableName2SchemaEvolutionPermission.containsKey(tableName)) {
+              this.tableName2SchemaEvolutionPermission.put(
+                  tableName,
+                  this.conn != null
+                      && this.conn.hasSchemaEvolutionPermission(
+                          tableName, connectorConfig.get(SNOWFLAKE_ROLE)));
+            }
+          });
```

Comment:
this should also be inside a an uber check of 
`enableSchematization` i.e populate the map only if `enableSchematization` = true

Reply:
Added.